### ISSUE TITLE
Add `ignoreFunctions: []` to `unit-disallowed-list`

### ISSUE THREAD

**Issue by pinkhominid** [stylelint/stylelint#6574]
### What steps are needed to reproduce the bug?

Use CSS functions that include a disallowed unit character inside a function that expects it, like calc.
```css
h2 {
  position: absolute;
  inset-inline-end: calc(50% - 100px);
}
```

### What Stylelint configuration is needed to reproduce the bug?

```json
{
  "rules": {
        "unit-disallowed-list": [
            ["px"],
            {
                ignoreFunctions: ["calc"]
            }
        ],
  }
}
```

### How did you run Stylelint?

```sh
stylelint **/*.css
```

### Which version of Stylelint are you using?

14.11.0

### What did you expect to happen?

CLI would finish with no errors

### What actually happened?

```sh
Invalid Option: Invalid option name "ignoreFunctions" for rule "unit-disallowed-list"
```

### Does the bug relate to non-standard syntax?

No

### Proposal to fix the bug

A solution similar to #5857

**Comment by ybiquitous** [comment]
@pinkhominid Thanks for the proposal. SGTM 👍🏼 

I've labeled the issue as **ready to implement**. Please consider [contributing](https://stylelint.io/contributing) if you have time.

There are [steps on how to add a new option](https://stylelint.io/developer-guide/rules#add-an-option-to-a-rule) in the Developer guide.

**Comment by jeddy3** [comment]
One note: like the [`ignoreFunctions: []` option in `unit-allowed-list`](https://stylelint.io/user-guide/rules/unit-allowed-list#ignorefunctions-function-regex-regexfunctionregexregex) this option will ignore _all_ units within the specified functions. It can't be limited to particular units. As such, the config would be:

```json
{
  "rules": {
    "unit-disallowed-list": [ ["px"], { "ignoreFunctions": ["calc"] }]
  }
}
```

(Which is different from https://github.com/stylelint/stylelint/issues/6574#issue-1529870436).

If you want to ignore specific units, then [a custom rule in a plugin](https://stylelint.io/developer-guide/plugins/) is probably a better fit.

**Comment by pinkhominid** [comment]
@jeddy3 thanks for the clarification. Ignoring all disallowed units is what we want for our use case. I've updated the config.

**Comment by mattxwang** [comment]
Hey all, quickly picked this up and submitted a PR in #6592!

**Comment by pinkhominid** [comment]
Thanks @mattxwang!
